server/src/main/java/org/apache/druid/server/coordination/ServerManager.java

```diff
@@ -195,18 +194,10 @@ public <T> QueryRunner<T> getQueryRunnerForSegments(Query<T> query, Iterable<Seg
     } else {
       return new ReportTimelineMissingSegmentQueryRunner<>(Lists.newArrayList(specs));
     }
+    Function<SegmentReference, SegmentReference> segmentMapFn = analysis.getDataSource().createSegmentMapFunction(query, cpuTimeAccumulator);
```

Comment:
does this need to be the `analysis.getDataSource()` instead of `query.getDataSource()`?

Reply:
I get confused between use of `analysis.getDataSource()` and `query.getDataSource()` especially when analysis is created from `DataSourceAnalysis.fromDataSource(query.getDataSource())`. This is the case here. I have updated it to use query

Reply:
Yes it is super easy to get confused by that...